src/OrderFormAcceptanceTests.Steps/Steps/CatalogueSolutions.cs

```diff
@@ -514,37 +511,49 @@ public void WhenTheUserRe_VisitsTheCatalogueSolution()
         [Then(@"the values will be populated with the values that was saved by the User")]
         public void ThenTheValuesWillBePopulatedWithTheValuesThatWasSavedByTheUser()
         {
-            if(Context.ContainsKey("AmendedDeliveryDate"))
+            if (Context.ContainsKey("AmendedDeliveryDate"))
             {
                 var expectedDate = (DateTime)Context["AmendedDeliveryDate"];
                 var dateValueFromPage = Test.Pages.OrderForm.GetProposedDate();
                 dateValueFromPage.Should().Be(expectedDate.ToString("dd MM yyyy"));
             }
 
-            if(Context.ContainsKey("AmendedEstimatedPeriod"))
+            if (Context.ContainsKey("AmendedEstimatedPeriod"))
             {
                 var expectedPeriod = (string)Context["AmendedEstimatedPeriod"];
                 var periodFromPage = Test.Pages.OrderForm.GetSelectedRadioButton();
                 periodFromPage.Should().Be(expectedPeriod);
             }
-                        
+
             var quantityFromPage = Test.Pages.OrderForm.GetQuantity();
-            var priceFromPage = Test.Pages.OrderForm.GetPriceInputValue();            
-            
+            var priceFromPage = Test.Pages.OrderForm.GetPriceInputValue();
+
             var expectedQuantity = (string)Context["AmendedQuantity"];
-            var expectedPrice = (string)Context["AmendedPrice"];           
-            
+            var expectedPrice = (string)Context["AmendedPrice"];
+
             quantityFromPage.Should().Be(expectedQuantity);
             priceFromPage.Should().Be(expectedPrice);
         }
 
         [Given(@"that the Supplier in the order has no associated services")]
+        public void GivenTheSupplierInTheOrderHasNoAssociatedServices()
+        {
+            var supplier = SupplierInfo.SuppliersWithout(Test.BapiConnectionString, CatalogueItemType.AssociatedService).First() ?? throw new NullReferenceException("Supplier not found");
+
+            var order = (Order)Context["CreatedOrder"];
+            order.SupplierId = int.Parse(supplier.SupplierId);
+            order.SupplierName = supplier.Name;
+            order.Update(Test.ConnectionString);
+        }
+
         [Given(@"the supplier added to the order has a solution with a declarative flat price")]
         public void GivenTheSupplierAddedToTheOrderHasASolutionWithADeclarativeFlatPrice()
         {
+            var supplier = SupplierInfo.SuppliersWithCatalogueSolution(Test.BapiConnectionString, ProvisioningType.Declarative).First() ?? throw new NullReferenceException("Supplier not found");
```

Comment:
given how often throw new NullReferenceException("Supplier not found"); is repeated, should we have a static method we can call?